src/Core/Migration/Migration1600349343AddDeliveryStateTransitions.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Migration\MigrationStep;
+use Shopware\Core\Framework\Uuid\Uuid;
+
+class Migration1600349343AddDeliveryStateTransitions extends MigrationStep
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1600349342;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $stateMachineId = $this->fetchOrderDeliveryStateId($connection);
+        $openStateId = $this->fetchOpenOrderDeliveryStateId($connection);
+        $missingStates = $this->fetchMissingOrderDeliveryStates($connection, $stateMachineId);
+
+        foreach ($missingStates as $missingState) {
+            $connection->insert(
```

Comment:
you should pre-validate if this not already exists

Reply:
Its calculated inside fetchMissingOrderDeliveryStates, all present ropen transition states are exluded.